### PR TITLE
callback extensions

### PR DESCRIPTION
**Pull request by singhal-ekansh**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -20,4 +25,23 @@ def ensure_an_admin_remains
         raise Error.new "Can't delete last user"
       end
     end
+
+    def ensure_not_delete_if_admin_email 
+      if self.email_was == 'admin@depot.com'
```
Comment (harsh-ag) [inline app/models/user.rb]: create a method `was_admin?` and use it at both places

File: app/models/user.rb [hunk 2]
```diff
@@ -4,6 +4,11 @@ class User < ApplicationRecord
   validates :name, presence: true, uniqueness: true
   has_secure_password
   
+  # callbacks extentions
+  before_destroy :ensure_not_delete_if_admin_email
```
Comment (harsh-ag) [inline app/models/user.rb:8]: rename method `check_admin_before_update` and similarly other

**Review by harsh-ag — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (singhal-ekansh) [reply app/models/user.rb]: done

File: app/models/user.rb (hunk 2 above)
Comment (singhal-ekansh) [reply app/models/user.rb:8]: done

**Review by harsh-ag — APPROVED** [review]

**Pull request closed without merging**